controllers/configurationpolicy_controller.go

```diff
@@ -1110,28 +1125,75 @@ func (r *ConfigurationPolicyReconciler) handleObjectTemplates(plc policyv1.Confi
 				relatedObjects = updateRelatedObjectsStatus(relatedObjects, object)
 			}
 		}
-		// violations for enforce configurationpolicies are already handled in handleObjects,
-		// so we only need to generate a violation if the remediationAction is set to inform
-		if !handled && !enforce {
-			objData := map[string]interface{}{
-				"indx":        indx,
-				"kind":        kind,
-				"desiredName": templateObjs[indx].name,
-				"namespaced":  templateObjs[indx].isNamespaced,
-			}
 
-			statusUpdateNeeded := createInformStatus(
+		objData := map[string]interface{}{
```

Comment:
Keeping this under the if statement would be more efficient since you wouldn't have to allocate memory for this map if the if condition is false.

Reply:
good!